sdk/core/Azure.Core/src/Request.cs

```diff
@@ -80,7 +80,11 @@ public virtual RequestUriBuilder Uri
         /// <param name="value">The header value.</param>
         protected internal virtual void SetHeader(string name, string value)
         {
+#if NETFRAMEWORK
```

Comment:
We shouldn't do this here. HttpClientTransport is used on .NET Framewowrk as well.